client/spring-zeebe-starter/src/main/java/io/camunda/zeebe/spring/client/SpringZeebeAutoConfiguration.java

```diff
@@ -0,0 +1,74 @@
+package io.camunda.zeebe.spring.client;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import io.camunda.zeebe.client.ZeebeClient;
+import io.camunda.zeebe.client.api.JsonMapper;
+import io.camunda.zeebe.client.impl.ZeebeObjectMapper;
+import io.camunda.zeebe.spring.client.actuator.ZeebeActuatorConfiguration;
+import io.camunda.zeebe.spring.client.annotation.customizer.ZeebeWorkerValueCustomizer;
+import io.camunda.zeebe.spring.client.annotation.processor.ZeebeAnnotationProcessorRegistry;
+import io.camunda.zeebe.spring.client.event.ZeebeLifecycleEventProducer;
+import io.camunda.zeebe.spring.client.properties.PropertyBasedZeebeWorkerValueCustomizer;
+import io.camunda.zeebe.spring.client.properties.ZeebeClientConfigurationProperties;
+import io.camunda.zeebe.spring.client.testsupport.SpringZeebeTestContext;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnMissingBean;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnProperty;
+import org.springframework.boot.context.properties.EnableConfigurationProperties;
+import org.springframework.context.ApplicationEventPublisher;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.context.annotation.Import;
+
+import java.lang.invoke.MethodHandles;
+
+/**
+ *
+ * ZeebeClientBuilder is provided by ZeebeClientStarterAutoConfiguration (which is enabled by META-INF of Spring Boot Starter)
+ *
+ * ZeebeClientObjectFactory can create ZeebeClients, it does that by being used as ZeebeClientLifecycle in AbstractZeebeBaseClientSpringConfiguration
+ */
+@Configuration
+@ConditionalOnProperty(prefix = "zeebe.client", name = "enabled", havingValue = "true",  matchIfMissing = true)
+@Import({
+  ZeebeClientConfiguration.class,
+  ZeebeActuatorConfiguration.class})
+@EnableConfigurationProperties(ZeebeClientConfigurationProperties.class)
+public class SpringZeebeAutoConfiguration extends AbstractZeebeBaseClientSpringConfiguration {
+
+  private static final Logger LOG = LoggerFactory.getLogger(MethodHandles.lookup().lookupClass());
+
+  private final ZeebeClientConfigurationProperties configurationProperties;
+  public SpringZeebeAutoConfiguration(ZeebeClientConfigurationProperties configurationProperties) {
+    this.configurationProperties = configurationProperties;
+  }
+
+  @Bean
+  @ConditionalOnMissingBean(SpringZeebeTestContext.class) // only run if we are not running in a test case - as otherwise the the lifecycle is controlled by the test
```

Comment:
I may be wrong, but it seems like defining the `SpringZeebeTestContext` bean is a bit redundant, because we already have `ZeebeClientTestConfiguration`, which is already only present in test context.

Reply:
True - but `ZeebeClientTestConfiguration` is not on the classpath in the starter - that's why I needed some class that I could create "up" in the module hierarchy